src/App.tsx

```diff
@@ -1,61 +1,124 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const prepearedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(user => user.id === todo.userId) || null,
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState([...prepearedTodos]);
+  const [userName, setUserName] = useState('');
+  const [isTitle, setIsTitle] = useState(true);
+  const [isUserName, setIsUserName] = useState(true);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title) {
+      setIsTitle(false);
+    }
+
+    if (!userName) {
+      setIsUserName(false);
+    }
+
+    if (title && userName) {
+      const isUser = usersFromServer
+        .find(user => user.name === userName) || null;
+
+      const newTodo: Todo = {
+        id: todos[todos.length - 1].id + 1,
+        title: title.trim(),
+        userId: isUser
+          ? isUser.id
+          : null,
+        completed: false,
+        user: isUser || null,
+      };
+
+      setTodos([...todos, newTodo]);
+      setTitle('');
+      setUserName('');
+    }
+  };
+
+  const handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setIsTitle(true);
+  };
+
+  const handleUserName = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserName(event.target.value);
+    setIsUserName(true);
+  };
 
-export const App = () => {
   return (
     <div className="App">
-      <h1>Add todo form</h1>
+      <h1 className="App__title">Add todo form</h1>
+
+      <form
+        className="App__form"
+        onSubmit={(event) => handleSubmit(event)}
```

Comment:
you can pass handler withouth explicit call, fix it everywhere
```suggestion
        onSubmit={handleSubmit}
```